spec/features/admin/timeline_batch_actions_spec.rb

```diff
@@ -0,0 +1,57 @@
+require 'rails_helper'
+
+feature 'timeline batch actions', js: true do
+  let(:division) { create(:division) }
+  let(:user) { create_member(division) }
+  let!(:loan) { create(:loan, :with_timeline, division: division) }
+
+  # Create a step with a dependent step, which also has a dependent step
+  let(:step1) { create(:project_step, project: loan) }
+  let(:step2) { create(:project_step, project: loan, scheduled_start_date: nil, schedule_parent: step1) }
+  let(:step3) { create(:project_step, project: loan, scheduled_start_date: nil, schedule_parent: step2) }
+
+  # Create two independent steps, one with and the other without a scheduled start date
+  let(:step4) { create(:project_step, project: loan, scheduled_start_date: Date.today, scheduled_duration_days: nil) }
+  let(:step5) { create(:project_step, project: loan, scheduled_start_date: nil, scheduled_duration_days: nil) }
+
```

Comment:
You can DRY the code below up a lot if you make an array with `let(:steps) { [step1, step2, ...] }` and then loop over it.